eventdemo/source/c/src/kaa_demo.c

```diff
@@ -19,212 +19,146 @@
 #include <string.h>
 #include <time.h>
 
-#include <kaa/kaa_error.h>
-#include <kaa/kaa_context.h>
-#include <kaa/platform/kaa_client.h>
-#include <kaa/utilities/kaa_log.h>
-#include <kaa/utilities/kaa_mem.h>
+#include <kaa_error.h>
+#include <kaa_context.h>
+#include <platform/kaa_client.h>
+#include <utilities/kaa_log.h>
+#include <utilities/kaa_mem.h>
 #include <kaa_user.h>
-#include <kaa/gen/kaa_thermostat_event_class_family.h>
+#include <gen/kaa_chat.h>
 
+#include <pthread.h>
 
 
-#define KAA_USER_ID            "user@email.com"
+#define KAA_USER_ID            "userid"
 #define KAA_USER_ACCESS_TOKEN  "token"
 
-#define THERMO_REQUEST_FQN          "org.kaaproject.kaa.schema.sample.event.thermo.ThermostatInfoRequest"
-#define CHANGE_DEGREE_REQUEST_FQN   "org.kaaproject.kaa.schema.sample.event.thermo.ChangeDegreeRequest"
+#define CHAT_EVENT_FQN     "org.kaaproject.kaa.examples.event.ChatEvent"
+#define CHAT_MESSAGE_FQN   "org.kaaproject.kaa.examples.event.Message"
 
 
+char current_room[100];
 
-static kaa_client_t *kaa_client = NULL;
-static bool is_shutdown = false;
-
-
-
-#define KAA_DEMO_UNUSED(x) (void)(x);
+struct room {
+    char room_name[100];
+    struct room *next;
+};
 
-#define KAA_DEMO_RETURN_IF_ERROR(error, message) \
-    if ((error)) { \
-        printf(message ", error code %d\n", (error)); \
-        return (error); \
-    }
-
-/*
- * Event callback-s.
- */
+typedef struct room Room;
 
-void kaa_on_thermostat_info_request(void *context
-                                  , kaa_thermostat_event_class_family_thermostat_info_request_t *event
-                                  , kaa_endpoint_id_p source)
+struct list
 {
-    KAA_DEMO_UNUSED(context);
-    KAA_DEMO_UNUSED(source);
-
-    printf("ThermostatInfoRequest event received!\n");
-
-    kaa_thermostat_event_class_family_thermostat_info_response_t *response =
-            kaa_thermostat_event_class_family_thermostat_info_response_create();
-
-    response->thermostat_info = kaa_thermostat_event_class_family_union_thermostat_info_or_null_branch_0_create();
-
-    kaa_thermostat_event_class_family_thermostat_info_t *info = kaa_thermostat_event_class_family_thermostat_info_create();
-    response->thermostat_info->data = info;
-
-    int32_t *current_degree = (int32_t *) KAA_MALLOC(sizeof(int32_t));
-    *current_degree = -5;
-
-    int32_t *target_degree = (int32_t *) KAA_MALLOC(sizeof(int32_t));
-    *target_degree = 10;
+    Room *head;
+    Room *tail;
+};
 
-    info->degree = kaa_thermostat_event_class_family_union_int_or_null_branch_0_create();
-    info->degree->data = current_degree;
+typedef struct list List;
 
-    info->target_degree = kaa_thermostat_event_class_family_union_int_or_null_branch_0_create();
-    info->target_degree->data = target_degree;
+List lst;
```

Comment:
This resource is shared between threads and access from different threads without locking may cause data races.